snowflake/objects/databases/recover/schemas/parquet/file_format/parquet_format.sql

```diff
@@ -0,0 +1,7 @@
+/*
+  Create the Parquet file format
+*/
+CREATE OR REPLACE FILE FORMAT {{ parquet_file_format_name }}
```

Comment:
It's a tradeoff.

`CREATE OR REPLACE` is the best way to ensure we are deploying in a declarative way without having to worry about state and requires minimal overhead. `FILE FORMAT` is a lightweight object and can be replaced easily.

But with [the caveat](https://docs.snowflake.com/en/sql-reference/sql/create-file-format#usage-notes) about `FILE FORMAT` and `EXTERNAL TABLE`, we either need to enforce top-down that external tables which use `FILE FORMAT` are always recreated during deployment, or – if we were to instead use `CREATE ... IF NOT EXISTS` – we need to depend on the developer to be aware that if they have an existing deployment and they modify the `FILE FORMAT` object, then those changes won't be reflected in subsequent deployments unless they manually delete the object. But if they manually delete the object, then they also have to manually deal with any fallout from the caveat I mentioned above! Alternatively, they could `ALTER` the file format, but already we're getting ourselves into a situation where the developer has to be familiar with and remember this caveat and they have to manage the state of those objects themselves; CI/CD won't do it for them. Of course, we'd have to deal with this complexity when deploying changes to a `FILE FORMAT` in staging or prod, too.

Fortunately, since `EXTERNAL TABLE` objects are merely metadata (schema and table properties), they are also lightweight objects and can be replaced easily. So my preference is to use `CREATE OR REPLACE` for both object types to ensure that we are deploying objects in a state that's reflected in their DDL.

Reply:
@philerooski Are we planning on using `EXTERNAL TABLE`?  That's an important caveat you mentioned - but if we aren't, it's probably ok to do this

Reply:
@thomasyu888 We are not, but if we were I would still recommend this approach because external tables are easy to replace.